### PR TITLE
Update zfs-events.5 with info from PSARC 2009/497

### PR DESCRIPTION
**Pull request by richardelling**
Update zfs-events.5 with info from PSARC 2009/497 regarding ereport fields.

Also updates ZIO_STAGE_* and ZIO_FLAG_* descriptions to match current source.

### Motivation and Context
Capture docs from original architectural docs.

### How Has This Been Tested?
manual man by man

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

Signed-off-by: Richard Elling <Richard.Elling@RichardElling.com>
Requires-builders: style

### CODE REVIEW

File: man/man5/zfs-events.5 [hunk 1]
```diff
@@ -768,7 +773,9 @@ Checksum value is byte swapped.
 \fBbad_ranges\fR
 .ad
 .RS 12n
-Checksum bad offset ranges.
+[start, end) pairs of corruption offsets. Offsets are always aligned on a
```
Comment (bunder2015) [inline man/man5/zfs-events.5]: mismatched brackets

Comment (richardelling) [reply man/man5/zfs-events.5]: this is mathematics set notation, not a programming language notation

File: man/man5/zfs-events.5 [hunk 2]
```diff
@@ -768,7 +773,9 @@ Checksum value is byte swapped.
...
+64-bit boundry, and can include some gaps of non-corruption.
```
Comment (bernie1995) [inline man/man5/zfs-events.5]: 64-bit bound**a**ry?

File: man/man5/zfs-events.5 [hunk 3]
```diff
@@ -777,7 +784,10 @@ Checksum bad offset ranges.
 \fBbad_ranges_min_gap\fR
 .ad
 .RS 12n
-Checksum allowed minimum gap.
+In order to bound the size of the bad_ranges array, gaps of non-corruption
+less than or equal to bad_ranges_max_gap bytes have been merged with adjacent
+corruption. Always at least 8, since corruption is detected on on a 64-bit
```
Comment (behlendorf) [inline man/man5/zfs-events.5]: s/at least 8,/at least 8 bytes,
s/on on/on

File: man/man5/zfs-events.5 [hunk 4]
```diff
@@ -768,7 +773,9 @@ Checksum value is byte swapped.
 \fBbad_ranges\fR
 .ad
 .RS 12n
-Checksum bad offset ranges.
+[start, end) pairs of corruption offsets. Offsets are always aligned on a
+64-bit boundary, and can include some gaps of non-corruption.
+(See bad_ranges_min_gap)
```
Comment (behlendorf) [inline man/man5/zfs-events.5]: `\fB`, `\fR` around the `bad_ranges_min_gap`.

File: man/man5/zfs-events.5 [hunk 5]
```diff
@@ -777,7 +784,10 @@ Checksum bad offset ranges.
 \fBbad_ranges_min_gap\fR
 .ad
 .RS 12n
-Checksum allowed minimum gap.
+In order to bound the size of the bad_ranges array, gaps of non-corruption
+less than or equal to bad_ranges_max_gap bytes have been merged with adjacent
```
Comment (behlendorf) [inline man/man5/zfs-events.5]: `\fB`, `\fR` around the `bad_ranges_max_gap`.

File: man/man5/zfs-events.5 [hunk 6]
```diff
@@ -786,7 +796,9 @@ Checksum allowed minimum gap.
 \fBbad_range_sets\fR
 .ad
 .RS 12n
-Checksum for each range the number of bits set.
+This array has one element per range in bad_ranges. Each element contains
```
Comment (behlendorf) [inline man/man5/zfs-events.5]: `\fB` `\fR` around `bad_ranges`.

File: man/man5/zfs-events.5 [hunk 7]
```diff
@@ -795,7 +807,9 @@ Checksum for each range the number of bits set.
 \fBbad_range_clears\fR
 .ad
 .RS 12n
-Checksum for each range the number of bits cleared.
+This array has one element per range in bad_ranges. Each element contains the
```
Comment (behlendorf) [inline man/man5/zfs-events.5]: `\fB` `\fR` around `bad_ranges`.

File: man/man5/zfs-events.5 [hunk 8]
```diff
@@ -804,7 +818,12 @@ Checksum for each range the number of bits cleared.
 \fBbad_set_bits\fR
 .ad
 .RS 12n
-Checksum array of bits set.
+If this field exists, it is an array of: (bad data & ~(good data)); that is,
+the bits set in the bad data which are cleared in the good data. Each element
+corresponds a byte whose offset is in a range in bad_ranges, and the array is
+ordered by offset. Thus the first element is the first byte in the first
+bad_ranges range and the last element is the last byte in the last bad_ranges
```
Comment (behlendorf) [inline man/man5/zfs-events.5]: `\fB` `\fR` around `bad_ranges`.
I'd suggests two commas:
  "Thus, the first...",
  "...range, and the last element..."

File: man/man5/zfs-events.5 [hunk 9]
```diff
@@ -813,7 +832,8 @@ Checksum array of bits set.
 \fBbad_cleared_bits\fR
 .ad
 .RS 12n
-Checksum array of bits cleared.
+Like bad_set_bits, but contains: (good data & ~(bad data)); that is, the bits
```
Comment (behlendorf) [inline man/man5/zfs-events.5]: `\fB` `\fR` around `bad_set_bits`.

**Comment by loli10K** [comment]
@richardelling can you please include the following change in this PR? It has been sitting in my TODO list for quite some time but i have never got around to pushing it; i would like to avoid opening a single byte PR unless required. Thanks!

```diff
diff --git a/man/man5/zfs-events.5 b/man/man5/zfs-events.5
index 4c60eecc5..3171b2581 100644
--- a/man/man5/zfs-events.5
+++ b/man/man5/zfs-events.5
@@ -406,7 +406,7 @@ Pool name.
 .RS 12n
 Failmode - \fBwait\fR, \fBcontinue\fR or \fBpanic\fR.
 See
-.BR pool (8)
+.BR zpool (8)
 (\fIfailmode\fR property) for more information.
 .RE


```

File: man/man5/zfs-events.5 (hunk 3 above)
Comment (richardelling) [reply man/man5/zfs-events.5]: done

File: man/man5/zfs-events.5 (hunk 5 above)
Comment (richardelling) [reply man/man5/zfs-events.5]: yep, also bad_ranges_max_gap is now bad_ranges_min_gap

File: man/man5/zfs-events.5 (hunk 6 above)
Comment (richardelling) [reply man/man5/zfs-events.5]: done

File: man/man5/zfs-events.5 (hunk 7 above)
Comment (richardelling) [reply man/man5/zfs-events.5]: done

File: man/man5/zfs-events.5 (hunk 8 above)
Comment (richardelling) [reply man/man5/zfs-events.5]: done

File: man/man5/zfs-events.5 (hunk 9 above)
Comment (richardelling) [reply man/man5/zfs-events.5]: done

**Pull request merged**